### PR TITLE
Extensions

### PR DESCRIPTION
**Pull request by iivchenko**
Little refactoring with extension methods.

Issue https://github.com/CrazyCouple/TActive/issues/9

### CODE REVIEW

File: Common/Common.Implementation/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        /// <summary>
+        /// Throws if a specified string is null, empty, or consists only of white-space characters.
+        /// </summary>
+        /// <param name="source">String under test.</param>
+        /// <param name="name">The name of the parameter that is under test.</param>
+        public static void ValidateEmpty(this string source, string name)
+        {
+            source.ValidateNull(name);
+
+            if (string.IsNullOrWhiteSpace(source))
```
Comment (mtarcha) [inline Common/Common.Implementation/Extensions/StringExtensions.cs:27]: For what you do 2 null validation here?
Don't you think that first validation "source.ValidateNull(name)" is excessive?

Comment (iivchenko) [inline Common/Common.Implementation/Extensions/StringExtensions.cs:27]: You're right sweetheart =)
I just want to divide exceptions: 
- first if the string is NULL -> ArgumentNullException
- second if the string is "" or "   " (let it be an 'empty string') -> ArgumentException

so I used DRY

Why do I did that?
I think that nulll and empty string could be caused by different use cases. Null means that program logic fail to get a real value and empty string means it is a real value but the value is not appropriate.

Comment (mtarcha) [inline Common/Common.Implementation/Extensions/StringExtensions.cs:27]: OK, you are right - but change the method name and add to xml doc. info about exceptions.

Comment (mtarcha) [inline Common/Common.Implementation/Extensions/StringExtensions.cs:27]: And it is not always true - "Null means that program logic fail to get a real value" =)

**Comment by mtarcha** [comment]
maladez =)

**Pull request merged**